src/libs/Navigation/Navigation.js

```diff
@@ -78,10 +85,19 @@ function closeDrawer() {
  * @returns {String}
  */
 function getDefaultDrawerState(isSmallScreenWidth) {
-    if (didTapNotificationBeforeReady) {
-        return 'closed';
+    if (didTapNotificationBeforeReady || !isSmallScreenWidth) {
+        return CONST.DRAWER_STATUS.CLOSED;
     }
-    return isSmallScreenWidth ? 'open' : 'closed';
+
+    const path = getPathFromState(navigationRef.current.getState(), linkingConfig.config).substring(1);
+
+    // Open the drawer when app directly open from root path
+    // and RHN modals are directly opened.
+    if (path === ROUTES.HOME || modal.isVisible === true) {
+        return CONST.DRAWER_STATUS.OPEN;
+    }
+
+    return CONST.DRAWER_STATUS.CLOSED;
```

Comment:
@parasharrajat I believe the screens inside `AuthScreen` have type modal except `SCREENS.HOME` is a drawer that contains `SidebarScreen` and `ReportScreen`.

https://github.com/Expensify/App/blob/7f7ad7741708a930ac9bba1764b1dbad76096edf/src/libs/Navigation/AppNavigator/AuthScreens.js#L179-L180